airflow/providers/amazon/aws/hooks/step_function.py

```diff
@@ -51,10 +53,26 @@ def start_execution(
             - :external+boto3:py:meth:`SFN.Client.start_execution`
 
         :param state_machine_arn: AWS Step Function State Machine ARN.
+        :param is_redrive_execution: Restarts unsuccessful executions of Standard workflows that did not
+            complete successfully in the last 14 days.
         :param name: The name of the execution.
         :param state_machine_input: JSON data input to pass to the State Machine.
         :return: Execution ARN.
         """
+        if is_redrive_execution:
+            if not name:
+                raise AirflowException(
+                    "Execution name is required to start RedriveExecution for %s.", state_machine_arn
+                )
```

Comment:
I think this should be `AirflowFailException` because if name isn't provided there is no value in allowing retry for the task

Reply:
Thank you @eladkal  good catch yes its not required for retry. I have updated now.